docs/sync-storage-handbook/src/places.md

```diff
@@ -0,0 +1,35 @@
+
+Places is based on the Firefox Desktop implementation, and uses a mostly backward-compatible storage schema. It can either be consumed directly, as in Firefox for iOS, or through a layer like [Mozilla Android Components](https://mozac.org/).
+
+## Architecture
+
+`PlacesApi` is the entry point to Places. It's expected that your app or component will create one `PlacesApi` instance, and use it as a singleton. `PlacesApi` manages database connections and global Sync state. You can use `PlacesApi::open_connection` to open multiple read-only connections, or acquire the single read-write connection. Syncing for the first time also opens a special read-write connection for Sync, but this connection is not exposed to applications.
+
+`open_connection` returns a `PlacesDb`. This is a wrapper around a [Rusqlite](https://docs.rs/rusqlite) `Connection` with shared `PRAGMA`s, SQL functions, and support for interrupting long-running queries. Under the hood, `PlacesDb` supports encrypted (via [SQLCipher](https://www.zetetic.net/sqlcipher/)) and unencrypted SQLite databases.
+
+When you're done with a connection, make sure to close it using `PlacesApi::close_connection`.
+
+## Connection lifecycle
+
+Opening a read-write connection to an existing database runs migrations to bring the database up to date. The current schema and version are compiled in to the Rust Places library. Opening a database with a newer schema version is also supported, but the version will be rolled back to the current version. Opening a connection to a file that doesn't exist creates and initializes an empty database at that path.
+
+In addition to the on-disk schema, a read-write connection also sets up temporary, in-memory tables and triggers. These are used internally for operations like deletion and frecency recalculation. The special Sync connection also defines its own tables and triggers to help with syncing. Read-only connections don't have these.
+
+All `PlacesDb` connections use SQLite's [WAL](https://sqlite.org/wal.html) mode. This is faster than the default rollback journal, and allows concurrent reads and writes. However, this does mean that read-only connections might not immediately see changes from the read-write or Sync connections. Reading occasionally stale data is fine for cases like autocomplete matching, where performance is critical above everything else. For other cases, such as populating history and bookmarks views, it's better to always read using the read-write connection. That way, reads after writes always return the just-written data.
```

Comment:
I thought that a new read starting after another write completes is guaranteed to still see the new data? If that's true, then this sounds a little pessimistic and may encourage people to never use the read-only connection because there's no guidance on how stale the data might be. I wonder if this can be dialed back a little to more general advice, such as "if you want to ensure that you are reading the same data which is possibly being concurrently written by the writer, then you should use the same write connection to read" or something similar?